awx_collection/plugins/modules/tower_user.py

```diff
@@ -130,18 +130,28 @@ def main():
     # Extract our parameters
     state = module.params.get('state')
     email = module.params.get('email')
+    password = module.params.get('password')
+    first_name = module.params.get('first_name')
+    last_name = module.params.get('last_name')
+    superuser = module.params.get('superuser')
+    auditor = module.params.get('auditor')
 
     # Create the data that gets sent for create and update
     user_fields = {
         'username': module.params.get('username'),
-        'first_name': module.params.get('first_name'),
-        'last_name': module.params.get('last_name'),
-        'password': module.params.get('password'),
-        'superuser': module.params.get('superuser'),
-        'auditor': module.params.get('auditor'),
     }
     if email is not None:
         user_fields['email'] = email
+    if password is not None:
+        user_fields['password'] = password
+    if first_name is not None:
+        user_fields['first_name'] = first_name
+    if last_name is not None:
+        user_fields['last_name'] = last_name
+    if superuser is not None:
+        user_fields['superuser'] = superuser
+    if auditor is not None:
+        user_fields['auditor'] = auditor
```

Comment:
aside from username, you should be able to just do this in a loop. But wouldn't block merging for it.